airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksDestination.java

```diff
@@ -60,8 +61,18 @@ public ExtendedNameTransformer getNameTransformer() {
   }
 
   @Override
-  public JdbcDatabase getDatabase(final JsonNode jsonConfig) {
-    return getDatabase(DatabricksDestinationConfig.get(jsonConfig));
+  public DataSource getDataSource(final JsonNode config) {
+    final DatabricksDestinationConfig databricksConfig = DatabricksDestinationConfig.get(config);
+    return DataSourceFactory.create(
```

Comment:
Good question.  Doesn't seem like it.  I am not too familiar with this code.  Ideally, we would want whatever runs the source/destination to close the DataSource once it completes.  For what it's worth, I don't believe the original `JdbcDatabase` was closed either.  I think that this would need to be redesigned in order to do so.

Reply:
I think the Databricks database is closed somewhere. Previously it was not properly closed, and the acceptance tests would just hang there.

Reply:
Confirmed that this is closed. The `getDatabase` method is used in `CopyDestination` in a `try` statement:

https://github.com/airbytehq/airbyte/blob/v0.38.2-alpha/airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/CopyDestination.java#L57

```java
    try (final JdbcDatabase database = getDatabase(config)) {
      final var nameTransformer = getNameTransformer();
      final var outputSchema = nameTransformer.convertStreamName(config.get(schemaFieldName).asText());
      AbstractJdbcDestination.attemptSQLCreateAndDropTableOperations(outputSchema, database, nameTransformer, getSqlOperations());

      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
    }
```

Reply:
> 

@tuliren That is still the case in the refactored code.  Instead of using the try-with-resources, it now uses a try/finally approach, as the Java `DataSource` interface unfortunately does not implement `AutoCloseable` or `Closeable`.  The areas that are being questioned are ones like this:

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksDestination.java#L41

The existing code gets a `Database` instance, but it is not clear that it is ever closed.  Perhaps it is eventually closed somewhere in the calling code?  I believe that this is the only place that @davinchia and I had questions about with regards to whether or not the changes in this PR match the existing behavior.

Reply:
@tuliren and I paired on this and we have resolved the issue with the DataSource not being closed.